roles/README.adoc

```diff
@@ -113,6 +113,16 @@ See also <<vars-vs-defaults,Vars vs Defaults>> which mandates "Avoid embedding l
 Since distribution-specific values are kind of "magic values", it applies to them.
 The same logic applies for providers: a role can load a provider-specific variable file, include a provider-specific task file, or both, as needed.
 Consider making paths to templates internal variables if you need different templates for different distributions.
+A role should have a single entry point. Requiring users to use `tasks_from: file.yml` is discouraged as this could cause important pre and post task files executed in `tasks/main.yml` to be skipped.
+The `state` or `action` keyword should be passed to a role when a role has such a capability and be required when calling the role. The `tasks/main.yml` should load the appropriate task file based on the action or state requested by the user, rather than the consumer having to have intimate knowledge of the file structure layout within the role.
+All facts added by a role should fall under a single fact key named as the role.
+All roles should use argspec validation, if multiple states or actions are supported argspec validation should be done with a task.
+All roles should use argspec validation at the egress to confirm facts are correctly set.
+The tasks folder or one of its subdirectories should not contain more than 10 files.
+All intermediate facts set during role execution should be prefaced with the role name.
+All roles should be packaged within an ansible collection
+Configuration management roles should be atomic, idempotent, and rollback when failures are encountered
+When many roles within a collection are similar, consider using a dispatcher role as the single entry point to invoke a specific role within the collection. An example of such may be a collection that does system health checks and contains many roles, each representing an individual health check.
```

Comment:
```suggestion
When many roles within a collection are similar, consider using a dispatcher role as the single entry point to invoke a specific role within the collection. 
An example of such may be a collection that does system health checks and contains many roles, each representing an individual health check.
```

Reply:
This example and recommendation is a curious to me. Is this that common a scenario? If not I'd recommend we leave it out especially if a tool like ansible-lint cannot easily identify the pattern and make a recommendation. My overarching concern with these guidelines is that they will overwhelm a would-be Ansible content developer from getting their heads wrapped around all of these and discourages them from even attempting to create something. 

Are these health checks going to be potentially run as one off operations? If not, then I question the value creating individual roles. This seems counter to my long standing understanding of roles being coarse-grain use case driven units of Ansible content and not programming libraries. Individual health checks sounds too fine-grain unless there is a compelling reasons why each individual health check may be executed independently with their own distinct handlers, defaults and vars. When I hear "dispatcher role" and "many similar roles within one collection" I feel like we are talking about writing coding and I wonder if the person who wrote that content is a Java programmer.

Reply:
@ssbarnea we need to be able to provide better documentation, examples, and rationale for suggestions like this.  I'm going to drop it from this PR, and it would need to be re-proposed as it's own change so that we don't drag out this PR if we want to keep it.